test/whitelisted_crowdsale.js

```diff
@@ -0,0 +1,75 @@
+"use strict";
+
+const MFC_Token = artifacts.require("MFC_Token");
+const WhitelistedCrowdsale = artifacts.require("WhitelistedCrowdsale");
+
+contract('WhitelistedCrowdsale', async (accounts) => {
+  let owner = accounts[0];
+  let user = accounts[1];
+
+  let now = Date.now();
+  let twoDays = 1000 * 60 * 60 * 24 * 2;
+
+  let startTime = now - twoDays;
+  let endTime = now + twoDays;
+  let rate = 30;
+  let minPurchase = 1;
+  let maxPurchase = 1000;
```

Comment:
Can we also add some tests with wrong parameters?
For example, maxPurchase = 1 and minPurchase = 1000 or startTime > endTime.
Not sure how it will behave, but ideally, it should fail with an (expected) error.

Reply:
In case when we use ICO_controller.startPrivateOffer for testing ICO_crowdsale constructor, we have to check only startTime, endTime, escrow . It will be even easier.

Reply:
I forgot say that in ICO_crowdsale constructor only values is assigning and there are no any checkings.

Reply:
@avgoeid OK, I see then I suggest to keep this test as it is. And add another task and test for `ICO_controller.startPrivateOffer` where we can also test assertions for min/max and start/end.